src/lib/equinoxe/equinoxe.ml

```diff
@@ -25,29 +25,78 @@
 include Equinoxe_intf
 
 (* Functor to build API using a specific call API system. *)
-module Make (B : Backend) : API = struct
-  open Json.Infix
+module Make (B : Backend) : API with type 'a io = 'a B.io = struct
+  type json = Ezjsonm.value
+  type 'a io = 'a B.io
 
-  type t = B.t
+  let return x = B.return x
+
+  (* let ( let+ ) m f = B.map f m *)
+  let ( let* ) m f = B.bind f m
+  let fail msg = B.fail (`Msg msg)
+
+  type t = { address : string; token : string }
 
   let create ?(address = "https://api.equinix.com/metal/v1/") ?(token = "") () =
-    B.create ~address ~token ()
+    { address; token }
+
+  module Http = struct
+    let url ~t ~path = Filename.concat t.address path
+
+    let headers ~token =
+      let token = if token = "" then [] else [ ("X-Auth-Token", token) ] in
+      token @ [ ("Content-Type", "application/json") ]
+
+    let get_json = function
+      | "" -> return (`O [])
+      | str -> (
+          let json = Ezjsonm.from_string str in
```

Comment:
This function can raise an `Ezjsonm.Parse_error`. Couldn't we add a mechanism to catch the error (`fail`)?